agent/hcp/client/client.go

```diff
@@ -1,4 +1,4 @@
-package hcp
+package client
```

Comment:
I considered using `hcpclient` as the package name, but the imports end up being named imports anyways i.e. : `import hcpclient "github.com/hashicorp/consul/agent/hcp/client"`

We'd ideally change the path itself to `/hcp/hcpclient` or something like that, but less repetitive to make this better! I was lacking imagination here so left it as client.

If anyone has any preferences, please let me know :)

Reply:
`client` is fine :)